src/main/java/com/bmc/truesight/meter/plugin/remedy/util/PluginTemplateValidator.java

```diff
@@ -34,11 +34,18 @@ public boolean validate(Template template) throws ValidationException {
                     new Object[]{payload.getTitle()}));
         }
 
-        // validate payload configuration
         for (String fpField : payload.getFingerprintFields()) {
-            if (fpField != null && fpField.startsWith("@") && !fieldItemMap.containsKey(fpField)) {
-                throw new ValidationException(
-                        StringUtil.format(Constants.PAYLOAD_PLACEHOLDER_DEFINITION_MISSING, new Object[]{fpField}));
+            if (fpField != null) {
+                if (fpField.startsWith("@")) {
+                    String field = fpField.substring(1);
+                    if (!Constants.FINGERPRINT_EVENT_FIELDS.contains(field)) {
+                        throw new ValidationException(StringUtil.format(Constants.EVENT_FIELD_MISSING, new Object[]{field, Constants.FINGERPRINT_EVENT_FIELDS}));
+                    }
+                } else {
+                    if (!payload.getProperties().containsKey(fpField)) {
+                        throw new ValidationException(StringUtil.format(Constants.EVENT_PROPERTY_FIELD_MISSING, new Object[]{fpField}));
```

Comment:
`Constants.EVENT_PROPERTY_FIELD_MISSING` also needs rewording

Reply:
May be we can reword it, it just suggests that the field does not exists in properties field of eventDefinition.